src/targets/gpu/kernels/include/migraphx/kernels/pad.hpp

```diff
@@ -29,34 +29,58 @@
 #include <migraphx/kernels/algorithm.hpp>
 #include <migraphx/kernels/ranges.hpp>
 #include <migraphx/kernels/vec.hpp>
+#include <migraphx/kernels/math.hpp>
 
 namespace migraphx {
 
-template <class Offsets, class Input, class Output, class PadVal>
+struct pad_constant
+{
+    constexpr diff_int operator()(diff_int idx, diff_int) const { return idx; }
+};
+struct pad_reflect
+{
+    constexpr diff_int operator()(diff_int idx, diff_int size) const
+    {
+        if(size == 1)
+            return 0;
+
+        auto period = size - 1;
```

Comment:
Is it fine to have a negative period here or is the intent so that period is always positive?

 Reading below with the remainder, it seems like you're using it as a modulo here which would give you negative mod val, but you could end up with a case if period is is negative, mod_val could also be negative. 
 
 I think your intent here was to have the output be positive because you're using `abs(idx)` which makes sense, otherwise the return value comparison would breakdown in some cases.